### DISCUSSION TITLE
EAN13 formatting of the text below the barcode.

### DISCUSSION THREAD

**Discussion by Woetoo** [micjahn/ZXing.Net#452] [Q&A] [answered]
This is actually two questions around a single topic.

Other EAN13 barcodes I've looked at have a very specific styling that I haven't been able to reproduce with ZXing yet. Perhaps it's not important to the actual scanning process - but it's an itch at the back of my mind I'm trying to scratch.

Firstly... The barcode lines I've seen elsewhere extend into the numbering area to create a sort of sectional/division zoning.
Secondly... The numbers tend to fit in specific areas within that bordering/zoning. With the first digit before the barcode and a ">" symbol after it.

Is this sort of styling possible with this iteration of Zxing?
![ean-13_1234567890128](https://user-images.githubusercontent.com/57496172/195064037-f13b91a6-8172-4283-9051-bb75f87452b5.jpg) -vs- ![ZXing_EAN13_1234567890128 ](https://user-images.githubusercontent.com/57496172/195065109-17795033-b3c2-4dd9-a720-985b0b80cc15.jpg)

I haven't seen anything in the wiki and experimenting with the `.options.` hasn't resolved it for me.

My second half of the main question...

Is it possible to change the font/font size of the text below the barcode.

I appreciate neither will affect the scanning of the barcode. This is purely me having a preconceived idea of what an EAN13 barcode looks like and trying to reproduce that before accepting that it really doesn't matter.

**Comment by micjahn** [comment, marked as answer]
You can try the following:
```
var writer = new BarcodeWriter
{
   Renderer = new ZXing.Rendering.AlternateBitmapRenderer();
}
```
The AlternateBitmapRenderer is not available for every .Net target platform / bitmap library binding.
If you are using .Net core or .Net 5.0 or above you have to use ZXing.Windows.Compatibility.AlternateBitmapRenderer from the ZXing.Net.Bindings.Windows.Compatibility package.

If the font doesn't fit your needs you can set a different one
```
var writer = new BarcodeWriter
{
   Renderer = new ZXing.Rendering.AlternateBitmapRenderer
   {
      TextFont = ...
   }
}
```

**Reply by Woetoo** [reply]
Cheers. I'm using VB .Net 6.

Got it mostly how I wanted it with `AlternateBitmapRender`. My testbench code current looks like:

```
Imports ZXing
Imports ZXing.Windows.Compatibility

        ' --- Later...

        Dim zxbc As BarcodeWriter = New BarcodeWriter
        zxbc.Format = BarcodeFormat.EAN_13
        zxbc.Renderer = New AlternateBitmapRenderer

        zxbc.Options.Height = PicBarcode2.Height
        zxbc.Options.Width = PicBarcode2.Width
        
        PicBarcode2.Image = New Bitmap(zxbc.Write("123456789012"))
```

`TextFont` seems like it's only available with in the standard renderer, but not the alternative version. But that's fine, as it was already resized how I imagined it should look.
The only thing I'm missing now is the "LMI" terminator (The ">" as text after the barcode).

Despite my newbie nature, cheers for the replies.

**Reply by micjahn** [reply]
There is no option to add the LMI.
